### ISSUE TITLE
thread.stop problem

### ISSUE THREAD

**Issue by levBagryansky** [objectionary/eo-threads#14]
It seems I cannot yet implement `thread.stop`.
This is implementation of `run` in `DataizingThread.java`:
```
    public Phi dataized() {
        return this.computed;
    }

    @Override
    public void run() {
        this.computed = new Data.ToPhi(new Dataized(this.eothread.attr("slow").get()).take());
    }
```
And stop in its dataization will do 
```
      final Phi parent = rho.attr("ρ").get();
      final DataizingThread thr = Threads.INSTANCE.get(parent);
      thr.interrupt();
      return new Data.ToPhi(true);
```
So in `DataizingThread.java` it must be 
```
    @Override
    public void run() {
        try {
            this.computed = new Data.ToPhi(new Dataized(this.eothread.attr("slow").get()).take());
        } catch (InterruptedException ie){
        }
    }
```
But compiler does not allow do that because
`Exception 'java.lang.InterruptedException' is never thrown in the corresponding try block`
So `Dataized.get` does not throw it. But Thread.sleep that is used in EOsleep.java can do.
@yegor256 @mximp may be we have other ways to stop the thread? I see only bad variants like deprecated `Thread.stop` and may be JNI. Or we need to mark some nethods in eo `throws InterruptedException`?

**Comment by levBagryansky** [comment]
@yegor256 please have a look

**Comment by yegor256** [comment]
@levBagryansky in Java there is not way to stop a thread unless the thread politely reacts to our signal. Read this, btw: https://www.yegor256.com/2015/10/20/interrupted-exception.html

So, I think, "do our best" and send the interrupted signal. If the thread stops -- great. If not, we can't do anything.

**Comment by yegor256** [comment]
@levBagryansky did you read this: https://www.yegor256.com/2015/10/20/interrupted-exception.html

This part:

```
while (/* You still need to wait */) {
    if (Thread.interrupted()) {
      throw new InterruptedException();
    }
    // Keep waiting
  }
```

**Comment by levBagryansky** [comment]
@yegor256 We need to handle InterruptedExeption in `get` method of `Atcompsite.java`

**Comment by 0pdd** [comment]
@levBagryansky the puzzle [#24](https://github.com/objectionary/eo-threads/issues/24) is still not solved.

**Comment by 0pdd** [comment]
@levBagryansky 2 puzzles [#24](https://github.com/objectionary/eo-threads/issues/24), [#45](https://github.com/objectionary/eo-threads/issues/45) are still not solved.

**Comment by 0pdd** [comment]
@levBagryansky all 2 puzzles are solved here: [#24](https://github.com/objectionary/eo-threads/issues/24), [#45](https://github.com/objectionary/eo-threads/issues/45).
